contracts/SaleVesting.sol

```diff
@@ -64,6 +72,16 @@ contract SaleVesting is Ownable {
         );
 
         for (uint256 i = 0; i < count; i++) {
+            //Check on the maximun size over which the for loop will run over.
+            require(i <= maxVestingDetailArray, "Too many vesting details");
+            //Check there are tokens available
```

Comment:
I see what you're doing after the audit report.  But this will be gas expensive.  Another approach could be:

```suggestion
//Check on the maximun size over which the for loop will run over.
        require(count <= maxVestingDetailArray, "Too many vesting details");

        for (uint256 i = 0; i < count; i++) {
            //Check there are tokens available
```